src/api/events/registerForEvents.js

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+import type { Auth, ApiResponseSuccess, Narrow } from '../apiTypes';
+import type { InitialData } from '../../types';
+import { apiPost } from '../apiFetch';
+
+/** See https://zulipchat.com/api/register-queue */
+
+type ApiResponseRegister = ApiResponseSuccess & {
+  queue_id: string,
+  last_event_id: number,
+  data: InitialData,
+};
+
+export default (
+  auth: Auth,
+  eventTypes: string[],
+  fetchEventTypes: string[],
+  applyMarkdown: boolean = true,
+  includeSubscribers: boolean = false,
+  clientGravatar: boolean = true,
+  allPublicStreams: boolean = false,
+  narrow: Narrow = [],
```

Comment:
This doesn't actually match the server API, according to the doc:

> A JSON-encoded array of length 2 indicating the narrow for which you'd like to receive events for. For instance, to receive events for the stream Denmark, you would specify `narrow=['stream', 'Denmark']`. Another example is `narrow=['is', 'private']` for private messages. Defaults to `"narrow=[]"`.

whereas our `Narrow` is

```
export type NarrowElement = {
  operand: string,
  operator?: NarrowOperator, // TODO type: this shouldn't be absent.
};

export type Narrow = NarrowElement[];
```

Better to leave this out than to get it wrong, and I think it's complex enough not to be worth trying to throw in until there's a use case. Ideally include a comment mentioning that there is such a thing and we just haven't wrapped it, e.g.:
```
  allPublicStreams: boolean = false,
  // also `narrow`, not currently wrapped here because it's a bit complicated
  // and we don't currently have a use for it.
```